src/main/java/net/openhft/chronicle/wire/WireMarshaller.java

```diff
@@ -486,13 +485,16 @@ public T defaultValue() {
     public void reset(T o) {
         try {
             for (FieldAccess field : fields) {
-                if (field.isResettable()) {
-                    Object value = field.field.get(o);
-                    if (value != null)
+                Class<?> resettableClass = field.resettableClass(defaultValue); // TODO FieldAccess should hold its defaultValue
+                if (resettableClass != null) {
+                    Object value = field.field.get(o); // TODO Replace with unsafe access
+                    if (value != null && resettableClass == value.getClass()) {
```

Comment:
if value instaceof Resettable call reset() on it?

Reply:
Makes sense changing the check to `Resettable`.